pandas/io/pytables.py

```diff
@@ -4789,7 +4789,7 @@ def _maybe_convert_for_string_atom(
 
         # we cannot serialize this data, so report an exception on a column
         # by column basis
-        for i in range(len(block.shape[0])):
+        for i in range(block.shape[0]):
```

Comment:
can you show what the `block.shape` is in the test suite now (pick a couple of example tests).